src/test/lib/DataTest.cpp

```diff
@@ -3556,6 +3553,263 @@ void QuicTestOperationPriority()
+            MsQuicStream Stream2(*Connections[1], QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpManual, ConnectionPriorityTestContext::ClientStreamStartStreamCallback, &Context);
+            MsQuicStream Stream3(*Connections[2], QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpManual, ConnectionPriorityTestContext::ClientStreamStartStreamCallback, &Context);
+            MsQuicStream Stream4(*Connections[3], QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpManual, ConnectionPriorityTestContext::ClientStreamStartStreamCallback, &Context);
+            MsQuicStream Stream5(*Connections[4], QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL, CleanUpManual, ConnectionPriorityTestContext::ClientStreamStartStreamCallback, &Context);
+
+            Stream1.Start(QUIC_STREAM_START_FLAG_IMMEDIATE);
+            // Wait until this StreamStart operation is drained
+            TEST_TRUE(Context.BlockAfterInitialStart.WaitTimeout(TestWaitTimeout));
+
+            TEST_QUIC_SUCCEEDED(Stream5.Start(QUIC_STREAM_START_FLAG_NONE));
+            TEST_QUIC_SUCCEEDED(Stream5.Send(&Buffer, 1, QUIC_SEND_FLAG_START | QUIC_SEND_FLAG_FIN));
+            TEST_QUIC_SUCCEEDED(Stream4.Start(QUIC_STREAM_START_FLAG_PRIORITY_WORK));
+            TEST_QUIC_SUCCEEDED(Stream4.Send(&Buffer, 1, QUIC_SEND_FLAG_START | QUIC_SEND_FLAG_FIN | QUIC_SEND_FLAG_PRIORITY_WORK));
+            TEST_QUIC_SUCCEEDED(Stream3.Start(QUIC_STREAM_START_FLAG_NONE));
+            TEST_QUIC_SUCCEEDED(Stream3.Send(&Buffer, 1, QUIC_SEND_FLAG_START | QUIC_SEND_FLAG_FIN));
+            TEST_QUIC_SUCCEEDED(Stream2.Start(QUIC_STREAM_START_FLAG_PRIORITY_WORK));
+            TEST_QUIC_SUCCEEDED(Stream2.Send(&Buffer, 1, QUIC_SEND_FLAG_START | QUIC_SEND_FLAG_FIN | QUIC_SEND_FLAG_PRIORITY_WORK));
+
+            TEST_QUIC_SUCCEEDED(Stream1.Send(&Buffer, 1, QUIC_SEND_FLAG_FIN | QUIC_SEND_FLAG_PRIORITY_WORK));
+
+            Context.OperationQueuedComplete.Set(); // All operations are queued. Kick off processing the operations
+            Context.MaxSend = 5;
+            TEST_TRUE(Context.AllReceivesComplete.WaitTimeout(TestWaitTimeout));
+
+            MsQuicStream* ExpectedStartOrder[5] = {0};
+            MsQuicStream* ExpectedSendOrder[5] = {0};
+            ExpectedStartOrder[0] = &Stream1;
+            ExpectedStartOrder[1] = &Stream4;
+            ExpectedStartOrder[2] = &Stream2;
+            ExpectedStartOrder[3] = &Stream5;
+            ExpectedStartOrder[4] = &Stream3;
+            ExpectedSendOrder[0] = &Stream1;
+            ExpectedSendOrder[1] = &Stream4;
+            ExpectedSendOrder[2] = &Stream2;
+            ExpectedSendOrder[3] = &Stream5;
+            ExpectedSendOrder[4] = &Stream3;
+
+            TEST_TRUE(memcmp(Context.StartOrder, ExpectedStartOrder, sizeof(ExpectedStartOrder)) == 0);
+            TEST_TRUE(memcmp(Context.SendOrder, ExpectedSendOrder, sizeof(ExpectedSendOrder)) == 0);
```

Comment:
Sometimes [3] and [4] are flipped. timing issue

Reply:
But how operations in FIFO queue swap.....

Reply:
Found the cause.
There is a case when [3] is in worker queue before explicitly enqueue operation by API.
Conventional enqueue doesn't re-enqueue the connection. Prioritized enqueue does.